### DISCUSSION TITLE
with or without AddTransient ExampleJob

### DISCUSSION THREAD

**Discussion by kzhui125** [quartznet/quartznet#1849] [Q&A]
I don't understand this doc:

// we can use options pattern to support hooking your own configuration
// because we don't use service registration api, 
// we need to manually ensure the job is present in DI
services.AddTransient<ExampleJob>();

It seems my job works without the AddTransient.

Can you help what's the purpose of using AddTransient?

Thanks!

**Comment by lahma** [comment]
So what are you doing instead?

**Reply by kzhui125** [reply]
I see the document use addtransient, If I don’t add this line of code, Will my job have problems?

So I want to know the purpose of using addtransient.

Thanks.

**Reply by kzhui125** [reply]
I forgort add this line of code: addtransient

**Reply by kzhui125** [reply]
If I don’t add this line of code: what will happen?

**Reply by lahma** [reply]
You probably don't need it if you don't have that use case.
